src/Features/OnboardingTasks.php

```diff
@@ -104,25 +96,55 @@ function( $gateway ) {
 		// task completion along with cache busting for active tasks.
-		$settings['onboarding']['automatedTaxSupportedCountries'] = self::get_automated_tax_supported_countries();
-		$settings['onboarding']['hasHomepage']                    = self::check_task_completion( 'homepage' ) || 'classic' === get_option( 'classic-editor-replace' );
-		$settings['onboarding']['hasPaymentGateway']              = ! empty( $enabled_gateways );
-		$settings['onboarding']['hasPhysicalProducts']            = count(
+		$settings['automatedTaxSupportedCountries'] = self::get_automated_tax_supported_countries();
+		$settings['hasHomepage']                    = self::check_task_completion( 'homepage' ) || 'classic' === get_option( 'classic-editor-replace' );
+		$settings['hasPaymentGateway']              = ! empty( $enabled_gateways );
+		$settings['hasPhysicalProducts']            = count(
 			wc_get_products(
 				array(
 					'virtual' => false,
 					'limit'   => 1,
 				)
 			)
 		) > 0;
-		$settings['onboarding']['hasProducts']                    = self::check_task_completion( 'products' );
-		$settings['onboarding']['isAppearanceComplete']           = get_option( 'woocommerce_task_list_appearance_complete' );
-		$settings['onboarding']['isTaxComplete']                  = self::check_task_completion( 'tax' );
-		$settings['onboarding']['shippingZonesCount']             = count( \WC_Shipping_Zones::get_zones() );
-		$settings['onboarding']['stylesheet']                     = get_option( 'stylesheet' );
-		$settings['onboarding']['taxJarActivated']                = class_exists( 'WC_Taxjar' );
-		$settings['onboarding']['themeMods']                      = get_theme_mods();
-		$settings['onboarding']['wcPayIsConnected']               = $wc_pay_is_connected;
+		$settings['hasProducts']                    = self::check_task_completion( 'products' );
+		$settings['isAppearanceComplete']           = get_option( 'woocommerce_task_list_appearance_complete' );
+		$settings['isTaxComplete']                  = self::check_task_completion( 'tax' );
+		$settings['shippingZonesCount']             = count( \WC_Shipping_Zones::get_zones() );
+		$settings['stylesheet']                     = get_option( 'stylesheet' );
+		$settings['taxJarActivated']                = class_exists( 'WC_Taxjar' );
+		$settings['themeMods']                      = get_theme_mods();
+		$settings['wcPayIsConnected']               = $wc_pay_is_connected;
+
+		return $settings;
+	}
+
+	/**
+	 * Add task items to component settings.
+	 *
+	 * @param array $settings Component settings.
+	 * @return array
+	 */
+	public function component_settings( $settings ) {
+		// Bail early if not on a wc-admin powered page, or task list shouldn't be shown.
+		if (
+			! \Automattic\WooCommerce\Admin\Loader::is_admin_page() ||
+			! \Automattic\WooCommerce\Admin\Features\Onboarding::should_show_tasks()
+		) {
+			return $settings;
+		}
+
+		// If onboarding isn't enabled this will throw warnings.
```

Comment:
Quick note in case this gets merged first- onboarding will always be enabled as of https://github.com/woocommerce/woocommerce-admin/pull/4999 but that probably only affects this comment and the check for `onboarding` being set is still a good measure.